packages/compass-home/src/modules/hashed-namespace.ts

```diff
@@ -0,0 +1,20 @@
+const HashedNamespace = new Map<string, Promise<string>>();
+
+export async function getHashedNamespace(ns: string): Promise<string> {
+  if (HashedNamespace.has(ns)) {
+    // We just checked that the property exists
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    return HashedNamespace.get(ns)!;
+  }
```

Comment:
I absolutely agree that this is annoying, but fwiw, if you want to avoid the assertion + linter silencing:

```suggestion
  const existingPromise = HashedNamespace.get(ns);
  if (existingPromise) {
    return existingPromise;
  }
```